pkg/test/utils/kubernetes.go

```diff
@@ -687,8 +691,34 @@ func FakeDiscoveryClient() discovery.DiscoveryInterface {
 				{
 					GroupVersion: corev1.SchemeGroupVersion.String(),
 					APIResources: []metav1.APIResource{
-						{Name: "pods", Namespaced: true, Kind: "Pod"},
-						{Name: "namespaces", Namespaced: false, Kind: "Namespace"},
+						{Name: "pod", Namespaced: true, Kind: "Pod"},
```

Comment:
Since fake discovery client is now needed in all the `workflow` package test, I extended it with the types we use in tests.